src/Controller/ModulesController.php

```diff
@@ -173,6 +173,165 @@ public function view($id) : ?Response
+                        $translation = $included;
+                    }
+                }
+            }
+            if (empty($translation)) {
+                throw new BEditaClientException(sprintf('Translation not found per %s %s and lang %s', $this->objectType, $id, $lang));
+            }
+        } catch (BEditaClientException $e) {
+            // Error! Back to index.
+            $this->log($e, LogLevel::ERROR);
+            $this->Flash->error($e, ['params' => $e->getAttributes()]);
+
+            return $this->redirect(['_name' => 'modules:list', 'object_type' => $this->objectType]);
+        }
+
+        $this->set('schema', $this->Schema->getSchema($this->objectType));
+
+        $object = Hash::extract($response, 'data');
+        $this->set('translation', $translation);
+        $this->set('object', $object);
+
+        return null;
+    }
+
+    /**
+     * Create or edit single translation.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function saveTranslation() : ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->prepareRequest($this->objectType);
+        $objectId = $requestData['object_id'];
+        if (!empty($requestData['id'])) {
+            unset($requestData['object_id']);
+        }
+        $lang = $requestData['lang'];
+        try {
+            $this->apiClient->save('translations', $requestData);
+        } catch (BEditaClientException $e) {
+            // Error! Back to object view or index.
+            $this->log($e, LogLevel::ERROR);
+            $this->Flash->error($e, ['params' => $e->getAttributes()]);
+
+            if ($this->request->getData('id')) {
+                return $this->redirect([
+                    '_name' => 'modules:translation',
+                    'object_type' => $this->objectType,
+                    'id' => $objectId,
+                    'lang' => $lang,
+                ]);
+            }
+
+            return $this->redirect([
+                '_name' => 'modules:createTranslation',
+                'object_type' => $this->objectType,
+                'id' => $objectId,
+            ]);
+        }
+
+        return $this->redirect([
+            '_name' => 'modules:translation',
+            'object_type' => $this->objectType,
+            'id' => $objectId,
+            'lang' => $lang,
+        ]);
+    }
+
+    /**
+     * Delete translation(s).
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function deleteTranslation() : ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->request->getData();
+        foreach ($requestData as $translation) {
+            if (!empty($translation['id'])) {
+                try {
+                    // remove completely the translation
+                    $this->apiClient->deleteObject($translation['id'], 'translations');
```

Comment:
we should deprecate `deleteObject` like `saveObject` in order to use `delete`, also valid for resources...